provider/azure/internal/imageutils/images.go

```diff
@@ -112,26 +112,41 @@ func SeriesImage(
 	}, nil
 }
 
+func offerForUbuntuSeries(series string) (string, string, error) {
+	seriesVersion, err := jujuseries.SeriesVersion(series)
+	if err != nil {
+		return "", "", errors.Trace(err)
+	}
+
+	oldSeries := set.NewStrings("trusty", "xenial", "bionic", "cosmic", "disco")
+	if oldSeries.Contains(series) {
+		return ubuntuOffering, seriesVersion, nil
+	}
+	seriesVersion = strings.ReplaceAll(seriesVersion, ".", "_")
+	return fmt.Sprintf("0001-com-ubuntu-server-%s", series), seriesVersion, nil
```

Comment:
I got the distinct impression from the email that the 0001 won't always be that value, and it will change as they update things. Does the ListSkus handle partial values?

Reply:
That 0001 doesn't come from ListSKUs(). The list api returns tags like "20_04-LTS" or "19_10" or "18.04-daily" etc.  That is one parameter used to select an image. The other is the offer string which is what the "0001-com-ubuntu-..." is. We need both the offer string and the SKU.